requests/models.py

```diff
@@ -796,6 +800,17 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             if delimiter:
                 lines = chunk.split(delimiter)
             else:
+                # Python splitlines() supports the universal newline (PEP 278).
+                # That means, '\r', '\n', and '\r\n' are all treated as end of 
+                # line. If the last chunk ends with '\r', and the current chunk
+                # starts with \n, they should be merged and treated as only 
+                # "one" new line separator '\r\n' by splitlines().
+                # This rule only applies when splitlines() is used.
+                if last_chunk_ends_with_cr and chunk.startswith(line_feed):
+                    # The last chunk ends with '\r', so the '\n' at chunk[0]
+                    # is just the second half of a '\r\n' pair rather than a
+                    # new line break. Just skip it.
+                    chunk = chunk[1:]
```

Comment:
One thing to keep in mind here is the final line of the file ending in `\r\n`. If we split at `\r`, we get a final one byte line of `\n`. This code makes chunk equivalent to `''` and `''.splitlines()` will result in an empty list. You'll get an index out of bounds with the call `lines[-1]` below.

Probably a rarer edge case but something we should cover.